src/components/extras/MapContainer.vue

```diff
@@ -18,88 +16,128 @@ export default {
-                this.$emit('location', this.marker.getLngLat());
+            // user marker styles
+            const pinScaled = new PinElement({
+                scale: 1.5,
+                // background: '#ffdd57',
+                // borderColor: '#0085ad',
+                // glyphColor: '#0085ad',
             });
-        }
-
-        // create DOM element for the parking site marker
-        for (const spots of this.spotsList) {
-            const psMarker = document.createElement('div');
-
-            psMarker.className = 'marker';
-            psMarker.style.backgroundImage = 'url(' + this.img + ')';
-            psMarker.style.width = '50px';
-            psMarker.style.height = '50px';
-            psMarker.style.backgroundSize = '110%';
-
-            const psPopup = new mapboxgl.Popup({ offset: 25 }).setHTML(
-                `<p><strong>${spots.Name}</strong></p><p><strong>Distance :</strong> ${spots.Distance} Km</p>`,
-            );
+            console.log('here', this.userLatLng);
+            // adding user marker in the map
+            this.userMarker = new AdvancedMarkerElement({
+                map: this.map,
+                position: this.userLatLng,
+                title: 'Your Location',
+                content: pinScaled.element,
+                gmpDraggable: this.drag, // make draggable marker
+            });
+            console.log(this.userMarker);
+            // Create an info window to share between markers.
+            const userInfoWindow = new InfoWindow();
+            this.userMarker.addListener('click', ({ domEvent, latLng }) => {
+                const { target } = domEvent;
+                console.log(target);
+                userInfoWindow.close();
+                userInfoWindow.setContent(this.userMarker.title);
+                userInfoWindow.open(this.userMarker.map, this.userMarker);
+            });
+            // when drag is true we can drag the marker
+            if (this.drag) {
+                this.userMarker.addListener('dragend', (event) => {
+                    const position = draggableMarker.position;
+                    userInfoWindow.close();
+                    userInfoWindow.setContent(
+                        `Pin dropped at: ${position.lat()}, ${position.lng()}`,
+                    );
+                    userInfoWindow.open(draggableMarker.map, draggableMarker);
+                });
+            }
 
-            new mapboxgl.Marker(psMarker)
-                .setLngLat([spots.Long, spots.Lat])
-                .setPopup(psPopup)
-                .addTo(this.map);
-        }
-    },
-    methods: {
-        ...mapMutations({
-            updateMapConfig: 'map/update-map-config',
-        }),
-        recenterMap(center) {
-            this.map.flyTo({
-                center: center,
+            this.spotsList.forEach((spot) => {
+                const contentString = `<div dir="ltr" style="" jstcache="0">
```

Comment:
instead of using HTMl as contentString can't we pass some component.

Reply:
Not in this scope